### ISSUE TITLE
misleading [assignment-from-none] warning

### ISSUE THREAD

**Issue by pwohlhart** [PyCQA/pylint#5952]
### Bug description

```
from typing import Optional


class Base(object):

  def get_name(self) -> Optional[str]:
    return None

  def print_message(self) -> None:
    name = self.get_name()
    print(f'This is my name: {name}')


class Derived(Base):

  def get_name(self) -> Optional[str]:
    return 'derived'


d = Derived()
d.print_message()
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
E1128: 13:Base.print_message: Assigning result of a function call, where the function returns None [assignment-from-none]
```

### Expected behavior

No warning. The return type is Optional[str] so pylint should figure that although the Base class always return None, derived classes might return a str.

### Pylint version

```shell
pylint: 2.7.4
astroid: 2.5.7+
logilab.common: 1.4.1
Python: 3.7.10
```

### OS / Environment

Debian

**Comment by mbyrnepr2** [comment]
This looks like a duplicate of https://github.com/PyCQA/pylint/issues/5503

**Comment by pwohlhart** [comment]
It does. Sorry I didn't see that one.
